### ISSUE TITLE
Interpolate(antialias=False) is 8X slower than antialias=True depending on the tensor stride

### ISSUE THREAD

**Issue by NicolasHug** [pytorch/pytorch#83840]
On a float32 tensor of shape `(3, 64, 64)` with stride `(4096, 64, 1)`, calling

```py
torch.nn.functional.interpolate(x, size=[224, 224], mode="bilinear", align_corners=False, antialias=antialias)
```

takes ~175 μs with `antialias=True ` and ~129μs with `antialias=False`. So far so good.

However, if we change the stride to `(1, 192, 3)` then `antialias=False` suddenly becomes 8X slower: 1471.8 μs (vs 176.4 μs for `antialias=True`, so fairly similar as before).

Perhaps this is expected but I was assuming that `antialias=False` is less work than `antialias=True`, and so we would expect it to be slightly faster?

CC @vfdev-5 @datumbox @nairbv 

### Reproducing example:

<details>

```py
import torch
from time import time

def bench(f, inp, num_exp=1000, num_prime=10):
    for _ in range(num_prime):
        f(inp)

    times = []
    for _ in range(num_exp):
        start = time()
        f(inp)
        end = time()
        times.append((end - start))
    
    median = torch.median(torch.tensor(times))
    print(f"Median over {num_exp} exp= {median * 1e6 :.1f} μs")
    return median


def interpolate(x, antialias):
    torch.nn.functional.interpolate(x, size=[224, 224], mode="bilinear", align_corners=False, antialias=antialias)


tensor_img = torch.randint(0, 256, (3, 64, 64), dtype=torch.float32)
assert tensor_img.stride() == (4096, 64, 1)
tensor_img = tensor_img[None, :, :, :]  # add batch dim for call to interpolate()

bench(lambda x: interpolate(x, antialias=True), tensor_img)  # 175.7 μs
bench(lambda x: interpolate(x, antialias=False), tensor_img)  # 129.7 μs
``` 

```py
tensor_img = torch.randint(0, 256, (3, 64, 64), dtype=torch.float32)
assert tensor_img.stride() == (4096, 64, 1)
tensor_img = tensor_img.as_strided(tensor_img.size(), stride=(1, 192, 3))
tensor_img = tensor_img[None, :, :, :]  # add batch dim for call to interpolate()

bench(lambda x: interpolate(x, antialias=True), tensor_img)  # 176.4 μs
bench(lambda x: interpolate(x, antialias=False), tensor_img)  # 1471.8 μs  8X slower!!!
```

</details>

## System Info

TL;DR: pytorch version == nightly 1.13.0.dev20220712

<details>
PyTorch version: 1.13.0.dev20220712
Is debug build: False
CUDA used to build PyTorch: 11.3
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.6 LTS (x86_64)
GCC version: (conda-forge gcc 9.5.0-16) 9.5.0
Clang version: 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
CMake version: version 3.22.3
Libc version: glibc-2.27

Python version: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:56:21)  [GCC 10.3.0] (64-bit runtime)
Python platform: Linux-5.4.0-1069-aws-x86_64-with-glibc2.27
Is CUDA available: True
CUDA runtime version: 11.0.221
GPU models and configuration: 
GPU 0: NVIDIA A100-SXM4-40GB
GPU 1: NVIDIA A100-SXM4-40GB

Nvidia driver version: 510.47.03
cuDNN version: Probably one of the following:
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.0.5
/usr/local/cuda-11.0/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.0.5
/usr/local/cuda-11.1/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.0.5
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_adv_infer.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_adv_train.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_cnn_infer.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_cnn_train.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_ops_infer.so.8.1.1
/usr/local/cuda-11.2/targets/x86_64-linux/lib/libcudnn_ops_train.so.8.1.1
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] numpy==1.22.3
[pip3] pytorch-pfn-extras==0.5.8
[pip3] torch==1.13.0.dev20220712
[pip3] torchdata==0.5.0a0+f1a128e
[pip3] torchvision==0.14.0a0+a35ef88
[conda] blas                      1.0                         mkl  
[conda] cudatoolkit               11.3.1               h2bc3f7f_2  
[conda] libblas                   3.9.0            14_linux64_mkl    conda-forge
[conda] libcblas                  3.9.0            14_linux64_mkl    conda-forge
[conda] liblapack                 3.9.0            14_linux64_mkl    conda-forge
[conda] liblapacke                3.9.0            14_linux64_mkl    conda-forge
[conda] mkl                       2022.0.1           h06a4308_117  
[conda] numpy                     1.22.3                   pypi_0    pypi
[conda] pytorch                   1.13.0.dev20220712 py3.9_cuda11.3_cudnn8.3.2_0    pytorch-nightly
[conda] pytorch-mutex             1.0                        cuda    pytorch-nightly
[conda] pytorch-pfn-extras        0.5.8                    pypi_0    pypi
[conda] torchdata                 0.5.0a0+f1a128e           dev_0    <develop>
[conda] torchvision               0.14.0a0+a35ef88           dev_0    <develop>

</details>

**Comment by vfdev-5** [comment]
@NicolasHug thanks for repro code ! Could you please also check if `interpolate(x, antialias=True)` case on `(1, 192, 3)` strides produces expected output ?

**Comment by NicolasHug** [comment]
@vfdev-5 done! I updated my message above and it's also on the reproducing example

**Comment by NicolasHug** [comment]
@vfdev-5 Sorry I misunderstood, I thought you asked for the execution time. I can confirm that the interpolation looks reasonable in all cases

<details>

```py
import torch
import torchvision.transforms as T
from torchvision.io import read_file, decode_jpeg
from PIL import Image
import matplotlib.pyplot as plt

file = "test/assets/encode_jpeg/grace_hopper_517x606.jpg"

from_pil = T.ToTensor()(Image.open(file))
assert from_pil.stride() == (313302, 517, 1)
from_decode = T.ConvertImageDtype(torch.float)(decode_jpeg(read_file(file)))
assert from_decode.stride() == (1, 1551, 3)

def interpolate(x, antialias):
    return torch.nn.functional.interpolate(x, size=[1000, 1000], mode="bilinear", align_corners=False, antialias=antialias)
    
    
for antialias in (True, False):
    for img in (from_pil, from_decode):
        img = img[None, :, :, :]  # add batch dim for call to interpolate()

        out = interpolate(img, antialias=antialias)
        fig = plt.figure()
        plt.imshow(out[0].permute(1, 2, 0))
plt.show()
```

</details>

**Comment by NicolasHug** [comment]
After chatting a bit with @fmassa ,[ `cpu_upsample_linear_channels_last` ](https://github.com/pytorch/pytorch/blob/1f29b3130af218847a043e58fdc64511bbe072fe/aten/src/ATen/native/cpu/UpSampleKernel.cpp#L1373) seems to be optimized for 4+ channels, but not for 3 channels (i.e. images), for which the generic kernel might be faster. A simple fix could then be:

```diff
diff --git a/aten/src/ATen/native/cpu/UpSampleKernel.cpp b/aten/src/ATen/native/cpu/UpSampleKernel.cpp
index 5f77bb7b57..e2e320ac34 100644
--- a/aten/src/ATen/native/cpu/UpSampleKernel.cpp
+++ b/aten/src/ATen/native/cpu/UpSampleKernel.cpp
@@ -1341,7 +1341,8 @@ void upsample_bilinear2d_kernel_impl(
     c10::optional<double> scales_w) {
 
   // Temporarily dispatch to original channels last implementation
-  if (input.is_contiguous(at::MemoryFormat::ChannelsLast)) {
+  if (input.is_contiguous(at::MemoryFormat::ChannelsLast) && (input.size(1) != 3)) {
     AT_DISPATCH_FLOATING_TYPES_AND(at::ScalarType::BFloat16, input.scalar_type(), "upsample_bilinear2d_channels_last", [&] {
       cpu_upsample_linear_channels_last<scalar_t, scale_t>(output, input, align_corners, {scales_h, scales_w});
     });
```

I'll try to run some benchmarks and submit a PR soon

**Comment by vfdev-5** [comment]
@NicolasHug we may also set the condition as `input.size(1) > 3`. There is another issue (tried to hacked in torchvision directly: https://github.com/pytorch/vision/pull/6661) with `input.size(1) == 1` which also takes channels last route.

**Comment by datumbox** [comment]
Reopening as the PR was reverted.
